cmd/dhcp-provider panics on startup

#### Issue Description

Reported by @StevenIsaacs in #250 and gitter
#### Steps to Reproduce (if applicable):

```
❯ ./dhcp-provider -c dhcp-provider.json
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x5612cd]

goroutine 1 [running]:
panic(0x909960, 0xc820012160)
    /usr/lib/go/src/runtime/panic.go:481 +0x3e6
github.com/cerana/cerana/vendor/github.com/spf13/viper.(*Viper).Get(0x0, 0x9bf1b0, 0xe, 0x0, 0x0)
    /home/mmendez/src/github.com/cerana/cerana/vendor/github.com/spf13/viper/viper.go:459 +0x7d
github.com/cerana/cerana/vendor/github.com/spf13/viper.(*Viper).GetDuration(0x0, 0x9bf1b0, 0xe, 0x0)
    /home/mmendez/src/github.com/cerana/cerana/vendor/github.com/spf13/viper/viper.go:570 +0x35
github.com/cerana/cerana/providers/dhcp.(*Config).LeaseDuration(0xc820013b00, 0x2)
    /home/mmendez/src/github.com/cerana/cerana/providers/dhcp/config.go:61 +0x3a
github.com/cerana/cerana/providers/dhcp.(*Config).Validate(0xc820013b00, 0x0, 0x0)
    /home/mmendez/src/github.com/cerana/cerana/providers/dhcp/config.go:29 +0x42
github.com/cerana/cerana/providers/dhcp.(*Config).LoadConfig(0xc820013b00, 0x0, 0x0)
    /home/mmendez/src/github.com/cerana/cerana/providers/dhcp/config.go:56 +0x67
main.main()
    /home/mmendez/src/github.com/cerana/cerana/cmd/dhcp-provider/main.go:36 +0x211
```

--- a/cmd/dhcp-provider/main.go
+++ b/cmd/dhcp-provider/main.go
@@ -9,6 +9,7 @@ import (
 	"github.com/cerana/cerana/provider"
 	"github.com/cerana/cerana/providers/dhcp"
 	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
 )
 
 func dieOnError(msg string, err error) {
@@ -26,13 +27,16 @@ func defaultNetwork() net.IPNet {
 func main() {
 	log.SetFormatter(&log.JSONFormatter{})
 
-	config := dhcp.NewConfig(nil, nil)
-	pflag.IP("gateway", nil, "default gateway")
-	pflag.Duration("lease-duration", 24*time.Hour, "default lease duration")
-	pflag.IPNet("network", defaultNetwork(), "network to manage dhcp addresses on")
-	pflag.StringSlice("dns-servers", nil, "dns servers")
-	pflag.Parse()
+	v := viper.New()
+	f := pflag.NewFlagSet("dhcp-provider", pflag.ExitOnError)
+	f.IP("gateway", nil, "default gateway")
+	f.Duration("lease-duration", 24*time.Hour, "default lease duration")
+	f.IPNet("network", defaultNetwork(), "network to manage dhcp addresses on")
+	f.StringSlice("dns-servers", nil, "dns servers")
 
+	config := dhcp.NewConfig(f, v)
+
+	dieOnError("error parsing args", f.Parse(os.Args))
 	dieOnError("error loading config", config.LoadConfig())
 	dieOnError("error setting up logging", config.SetupLogging())
 

